numpy/core/arrayprint.py

```diff
@@ -521,6 +509,8 @@ def array2string(a, max_line_width=None, precision=None,
     if formatter is None:
         formatter = _formatter
 
+    if style is not None and a.shape == () and a.dtype.names is None:
+        return style(a[()])
```

Comment:
> Do you mind leaving this point for consideration in another PR?

I've folded those into #9201. There's code in the wild that wants to treat `np.array('Some string')` as if it were `'Some string'` - so really, we shouldn't allow `setstringfunction` to change that